packages/tokenbank_v1/contracts/modules/security/RecoveryModule.sol

```diff
@@ -144,30 +149,26 @@ contract RecoveryModule is SecurityModule
         view
         returns (address[] memory signers)
     {
-        uint skipBytes = 0;
-        if (method == this.startRecovery.selector) {
-            // data layout: {length:32}{sig:4}{_wallet:32}{_newOwner:32}{signer1:32}{signer2:32}{...}
-            skipBytes = 68; // 4+32+32
-        } else if (method == this.cancelRecovery.selector) {
-            // data layout: {length:32}{sig:4}{_wallet:32}{signer1:32}{signer2:32}{...}
-            skipBytes = 36; // 4+32
-        } else if (method == this.completeRecovery.selector){
-            // data layout: {length:32}{sig:4}{_wallet:32}
-            skipBytes = 36; // 4+32
-            require(data.length == skipBytes, "INVALID_DATA");
-        }
-
-        require(
-            data.length >= skipBytes && (data.length - skipBytes) % 32 == 0,
-            "DATA_INVALID"
+        require (
+            method == this.startRecovery.selector ||
+            method == this.cancelRecovery.selector ||
+            method == this.completeRecovery.selector,
+            "INVALID_METHOD"
         );
 
-        uint numSigners = (data.length - skipBytes) / 32;
+        // ASSUMPTION:
+        // data layout: {data_length:32}{wallet:32}{signers_length:32}{signer1:32}{signer2:32}
```

Comment:
Arrays are ABI encoded slightly differently with an offset to the data:
- https://solidity.readthedocs.io/en/v0.5.13/abi-spec.html#examples

So I think a call to e.g. `completeRecovery` needs data like this:
```
{wallet:32}{signers_offset:32}{signers_length:32}{signer1:32}{signer2:32}
```

Reply:
This is really helpful. Please take another look.